### PR TITLE
Extensible FlattenStrategy.

### PR DESCRIPTION
**Pull request by andersio**
In case we might want to add new flatten strategies in the future without a major version bump, `FlattenStrategy` now hides its enum nature. This on paper breaks switches over `FlattenStrategy`, but I don't see any real use case that would be affected.

#### Checklist
- ~~Updated CHANGELOG.md.~~

### CODE REVIEW

File: Sources/Flatten.swift [hunk 1]
```diff
@@ -53,16 +73,21 @@ public enum FlattenStrategy: Equatable {
...
 			return true
 
 		case (.concurrent(let leftLimit), .concurrent(let rightLimit)):
 			return leftLimit == rightLimit
 
+		case (.race, .race):
+			return true
+
 		default:
 			return false
```
Comment (NachoSoto) [inline Sources/Flatten.swift]: Not saying I agree with this change, but if we merge it, it doesn’t make sense to keep the `Equatable` conformance.

File: Sources/Flatten.swift [hunk 2]
```diff
@@ -9,20 +9,36 @@
...
+		self.base = base
+	}
+
 	/// The producers should be merged, so that any value received on any of the
 	/// input producers will be forwarded immediately to the output producer.
 	///
 	/// The resulting producer will complete only when all inputs have
 	/// completed.
-	public static let merge = FlattenStrategy.concurrent(limit: .max)
+	public static var merge: FlattenStrategy {
```
Comment (NachoSoto) [inline Sources/Flatten.swift]: This doesn’t need to be a derived property, why not keep a constant?

**Comment by NachoSoto** [comment]
Can you elaborate on the “without a major version bump” thing? Is that in case users are `switch`ing over it?

**Comment by andersio** [comment]
Yes, that's the reason @mdiep blocked #233 from merging into 1.1.x.

**Comment by NachoSoto** [comment]
Cool then I’m on board after those 2 comments

**Comment by mdiep** [comment]
I'll review soon, but this is a great idea. 💯

**Pull request merged**